lib/blocks.php

```diff
@@ -93,3 +93,85 @@ function do_blocks( $content ) {
 	return $content_after_blocks;
 }
 add_filter( 'the_content', 'do_blocks', 9 ); // BEFORE do_shortcode() and wpautop().
+
+/**
+ * Extract the blocks from post content for the REST API post response.
+ *
+ * @since 1.1.0
+ *
+ * @param string The post content.
+ *
+ * @return array Array of block data.
+ */
+function get_block_data_for_api_from_post_content( $content ) {
+	$registry = WP_Block_Type_Registry::get_instance();
+	$blocks   = gutenberg_parse_blocks( $content );
+	$data     = array();
+
+	// Loop thru the blocks, adding rendered content when available.
+	foreach ( $blocks as $block ) {
+		$block_name  = isset( $block['blockName'] ) ? $block['blockName'] : null;
+		$attributes  = is_array( $block['attrs'] ) ? $block['attrs'] : null;
+		$raw_content = isset( $block['rawContent'] ) ? $block['rawContent'] : null;
+		if ( null !== $block_name ) {
+			$block_type = $registry->get_registered( $block_name );
+			if ( null !== $block_type ) {
+				$block['renderedContent'] = $block_type->render( $attributes, $raw_content );
+			}
+
+			// Set up the item data.
+			$item_data = array();
+			$item_data['type'] = $block['blockName'];
+			if ( null !== $attributes ) {
+				$item_data['attributes'] = $block['attrs'];
+			}
+			$item_data['content'] = $block['rawContent'];
```

Comment:
`$block['rawContent']` -> `$raw_content`

Also, should this only be exposed for `$context === 'edit'`?

Reply:
This is perhaps misnamed when it is returned from the parser - its not really the raw content and doesn't include the html markers for example. This really represents the rendered html and is returned as `content` for a block (matching naming from https://github.com/WordPress/gutenberg/pull/2503). As far as I can see, `rendered` is only used for a couple of block types where their content is dynamic, for example a recent post list.

Reply:
Can you review the data thats output, it seems public? we could show blocks only in edit context, but that would make my idea of using them for a front end data source less useful.